remme/clients/pub_key.py

```diff
@@ -29,31 +31,33 @@
 from cryptography.hazmat.backends import default_backend
 
 
-class CertificateClient(BasicClient):
+class PubKeyClient(BasicClient):
     def __init__(self, test_helper=None):
-        super().__init__(CertificateHandler, test_helper=test_helper)
+        super().__init__(PubKeyHandler, test_helper=test_helper)
 
     @classmethod
-    def get_new_certificate_payload(self, certificate_raw, signature_rem, signature_crt, cert_signer_public_key):
-        payload = NewCertificatePayload()
-        payload.certificate_raw = certificate_raw
-        payload.signature_rem = signature_rem
-        payload.signature_crt = signature_crt
-        if cert_signer_public_key:
-            payload.cert_signer_public_key = cert_signer_public_key
+    def get_new_pub_key_payload(self, public_key, entity_hash, entity_hash_signature, valid_from, valid_to):
+        payload = NewPubKeyPayload()
+        payload.public_key = public_key
+        payload.public_key_type = NewPubKeyPayload.RSA
+        payload.entity_type = NewPubKeyPayload.PERSONAL
+        payload.entity_hash = entity_hash
+        payload.entity_hash_signature = entity_hash_signature
+        payload.valid_from = valid_from
+        payload.valid_to = valid_to
 
         return payload
 
     @classmethod
     def get_revoke_payload(self, crt_address):
-        payload = RevokeCertificatePayload()
+        payload = RevokePubKeyPayload()
         payload.address = crt_address
 
         return payload
 
-    def process_csr(self, certificate_request, key, not_valid_before=None, not_valid_after=None):
-        public_key = certificate_request.public_key()
-        subject = certificate_request.subject
+    def process_csr(self, pub_key_request, key, not_valid_before=None, not_valid_after=None):
```

Comment:
this shouldn't be changed. 
we still can have csr for certificate in client.

Reply:
It do not changed. Only name of variable

Reply:
There are no pub_key_requests. only cert_requests https://en.wikipedia.org/wiki/Certificate_signing_request